server/chat-plugins/scavenger-games.ts

```diff
@@ -213,6 +220,46 @@ const TWISTS: {[k: string]: Twist} = {
 		},
 	},
 
+	spamfilter: {
+		name: 'Spam Filter',
+		id: 'spamfilter',
+
+		desc: 'Every wrong answer adds 30 seconds to your final time!',
+
+		onIncorrectAnswer(player: ScavengerHuntPlayer, value: string) {
+			if (!player.incorrect) player.incorrect = [];
+			const id = `${player.currentQuestion}-${value}`;
+			if (player.incorrect.includes(id)) return;
+
+			player.incorrect.push(id);
+		},
+
+		onComplete(player, time, blitz) {
+			const seconds = toSeconds(time);
+			if (!player.incorrect) return {name: player.name, total: seconds, blitz, time, original_time: time};
+
+			const deduction = 30 * player.incorrect.length;
+			const total = seconds + deduction;
+			const final_time = Chat.toDurationString(total * 1000, {hhmmss: true});
+			if (total > 60) blitz = false;
+
+			return {name: player.name, total, blitz, time: final_time, original_time: time};
+		},
+
+		onConfirmCompletion(player, time, blitz, place, result) {
+			blitz = result.blitz;
+			time = result.time;
+			const deduction_message = player.incorrect?.length ?
+				`${player.incorrect.length} incorrect ${Chat.plural(player.incorrect.length, "guesses", "guess")}` :
+				"Perfect!";
+			return `<em>${Utils.escapeHTML(player.name)}</em> has finished the hunt! (Final Time: ${time} - ${deduction_message}${(blitz ? " - BLITZ" : "")})`;
+		},
+
+		onEnd() {
+			this.completed.sort((a, b) => a.total - b.total);
```

Comment:
```suggestion
			Utils.sortBy(this.completed, entry => -entry.total)
```

Reply:
I don't agree with this one.  It seems like a lot of extra steps looking at utils file. a.total is always type number, so there's no reason to use Utils which has a lot more type definition in it's comparative process.

Reply:
The Utils differences don't matter much. This is just more legible / consistent (since sortBy is preferred)